### ISSUE TITLE
Ordering of filters and weight of http features

### ISSUE THREAD

**Issue by re-thc** [avaje/avaje-http#503]
Hi, in using `@filter` is there a way to specify the order? The helidon docs says that they get applied based the order they're added to Helidon. Is that then based on how we add the beans / HttpFeature? There's no way at the moment to sort it in some specific order.

More generally speaking, filters and controllers create Http Features, which have weights that impact ordering. Is there any way to set this?

Also is each controller a HttpFeature? Can we group things?

Thanks!

**Comment by SentryMan** [comment]
Each controller is indeed a http feature, with each route within the controller registered in order of appearance.

**Comment by SentryMan** [comment]
Weight of controllers rarely matter, as filters always go before routes

**Comment by rbygrave** [comment]
Yes, we will need a way to order `@Filter`s and that might be missing. For avaje-inject there is a `@Priority` and ability to get a list of components ordered by the priority value so that might be all that we are after at this stage.

For avaje-nima when adding HttpFeatures we'd change that from using `beanScope.list(HttpFeature.class)` to `beanScope.listByPriority(HttpFeature.class)`.

**Comment by rbygrave** [comment]
Perhaps we are over thinking it in that the simplest approach is to have a dedicated controller for filters [and perhaps exceptions] and the order of the filter methods there is the order in which they are registered.

For example:
```java
@Controller
final class ErrorController {

  @ExceptionHandler(statusCode = 407)
  Map<String, Object> runEx(RuntimeException ex, ServerRequest req, ServerResponse res) {
    return Map.of("err", String.valueOf(ex));
  }

  @Filter
  void filter1(FilterChain chain, RoutingResponse res) {
    chain.proceed();
  }

  @Filter
  void filter2(FilterChain chain, RoutingResponse res) {
    chain.proceed();
  }

  @Filter
  void filter(FilterChain chain, RoutingResponse res) {
    chain.proceed();
  }
}
```
generates ...

```java
  @Override
  public void setup(HttpRouting.Builder routing) {
    routing.error(RuntimeException.class, this::_runEx);
    routing.addFilter(this::_filter1);
    routing.addFilter(this::_filter2);
    routing.addFilter(this::_filter);
  }
```
... so the order of the filters registered just matches the order in which they are defined in source.

**Comment by SentryMan** [comment]
> Perhaps we are over thinking it in that the simplest approach is to

Closing unless more details are added for why the above wouldn't work
